### PR TITLE
Enable permissive mode for default deployment

### PR DESCRIPTION
**Pull request by wattli**
This change applies to istio-demo.yaml 

All services will have permissive mode (they can take both http and mtls) now. But clients still send http traffic. 

Will have separate PRs to remove istio-demo-auth.yaml

### CODE REVIEW

**Comment by rshriram** [comment]
shouldn't this go to release-1.0 ?

**Comment by linsun** [comment]
Does this mean we'll have one single install for quick start using yaml?  e.g. no need to recommend users to install istio demo auth yaml file?

Given this PR is for user deployed microservices onto istio, how are we enabling user to config permissive mode  for the control plane itself?

File: tests/e2e/tests/pilot/authn_policy_test.go [hunk 1]
```diff
@@ -27,21 +27,12 @@ func TestMTlsWithAuthNPolicy(t *testing.T) {
...
 		t.Skipf("Skipping %s: authn=true", t.Name())
 	}
-	// This policy will enable mTLS globally (mesh policy)
-	globalCfg := &deployableConfig{
-		Namespace:  "", // Use blank for cluster CRD.
-		YamlFiles:  []string{"testdata/authn/v1alpha1/global-mtls.yaml.tmpl"},
-		kubeconfig: tc.Kube.KubeConfig,
-	}
-	// This policy disable mTLS for c and d:80.
+	// This policy enable mTLS for c and d:80.
```
Comment (tiswanso) [inline tests/e2e/tests/pilot/authn_policy_test.go]: Need to finish removal of globalCfg.  e2e tests are failing to build.

**Comment by costinm** [comment]
I don't think this should be in 1.0.1 or 1.0 - we will need a long testing time, and it's a big behavior change.

Please also open a thread on istio-dev, announcing the intention and linking to the PR - this needs maximum visibility.  I think it is a great change for both increasing security and simplifying istio.

TODO items (possibly in separate PR - maybe open tracking issues so we don't miss it): 
- if a workload doesn't have certificates ( files missing ) - make sure plain text is used instead of failing.
- if a workload has certificates - use the secure port for pilot, mixer (control plane policy).
- add at least one test where an explicit policy sets 'plain text' - so we know users can control this. I believe we have tests where we set STRICT mode.
- remove the duplicated 'auth' suites !!
- remove istio-demo-auth
- scrub the docs for the istio-demo-auth

**Comment by rshriram** [comment]
@costinm this is an upgrade issue. Anyone who has a strict mtls setting (which is the default in 1.0) will be surprised to find out that non mtls traffic is going through. Thats not good to change in 1.1, esp when someone has this running in prod. So, we either change this now in 1.0 or we don't change at all, because it will break installations that *assume* strict is the default mode.

File: install/kubernetes/helm/istio/charts/security/templates/enable-mesh-permissive.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  labels:
+    app: istio-security
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  peers:
+  - mtls:
+      mode: PERMISSIVE
+{{- end }}
```
Comment (baodongli) [inline install/kubernetes/helm/istio/charts/security/templates/enable-mesh-permissive.yaml]: this file is the same as install/kubernetes/helm/istio-remote/charts/security/templates/enable-mesh-permissive.yaml. A symlink pointing to the same file should have worked. But it might be done in a different PR.

File: tests/e2e/tests/pilot/authn_policy_test.go [hunk 3]
```diff
@@ -27,25 +27,15 @@ func TestMTlsWithAuthNPolicy(t *testing.T) {
 		// The whole authn test suites should be rewritten after PR #TBD for better consistency.
 		t.Skipf("Skipping %s: authn=true", t.Name())
 	}
-	// This policy will enable mTLS globally (mesh policy)
-	globalCfg := &deployableConfig{
```
Comment (diemtvu) [inline tests/e2e/tests/pilot/authn_policy_test.go]: It's better to use option from https://github.com/istio/istio/pull/7778 to 'modify' the global policy rather than bend the test. We want to keep at least one test for the global mTLS enablement.

File: tests/e2e/tests/pilot/testdata/authn/v1alpha1/authn-policy.yaml.tmpl [hunk 4]
```diff
@@ -6,5 +6,5 @@ spec:
   targets:
   - name: "c"
   - name: "d"
-    ports:
-    - number: 80
+  peers:
+  - mtls: {}
```
Comment (baodongli) [inline tests/e2e/tests/pilot/testdata/authn/v1alpha1/authn-policy.yaml.tmpl]: The name of this rule indicates 'disable mtls'. But is it actually for enabling mTLS for 'c' and 'd'?

Comment (diemtvu) [reply tests/e2e/tests/pilot/testdata/authn/v1alpha1/authn-policy.yaml.tmpl]: +1. Name should be updated to be consistent with the new intention. Another reason for not changing the test set up.

Comment (wattli) [reply tests/e2e/tests/pilot/testdata/authn/v1alpha1/authn-policy.yaml.tmpl]: @diemtvu , i can restore the existing policy setup for this case, but there are many test cases we do want to update, check the failing tests. https://circleci.com/gh/istio/istio/155866?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Comment (diemtvu) [reply tests/e2e/tests/pilot/testdata/authn/v1alpha1/authn-policy.yaml.tmpl]: It's probably a true problem. One of the failure test is  TestGateway_TCPIngress/TCPIngressGateway. Do you think PERMISSIVE could do smt wrong for TCP?

**Comment by costinm** [comment]
We only ship a demo file with strict mode - the helm install defaults to
plain text, and a user who has the set value to strict should get the same
when upgrading

**Comment by costinm** [comment]
I mean - istio 1.0 doesn't ship any istio-auth.yaml, but a istio-demo, not
intended for prod.

What we must test is that if the helm option to enable strict has been set,
it will be honored after the change

**Comment by wattli** [comment]
This matches my plan, will have follow-up PRs for this.
